graylog2-server/src/main/java/org/graylog2/indexer/searches/SearchesConfig.java

```diff
@@ -87,10 +87,19 @@ public static Builder builder() {
 
         public abstract Builder limit(int limit);
 
+        public abstract int limit();
+
         public abstract Builder offset(int offset);
 
         public abstract Builder sorting(Sorting sorting);
 
-        public abstract SearchesConfig build();
+        abstract SearchesConfig autoBuild();
+
+        public SearchesConfig build() {
+            if (limit() == 0) {
```

Comment:
Good point, I guess only if someone supplies `< 0` via the API, which is nonsensical anyway.

I'll update this and the test to provide a sensible default.